--- py/selenium/webdriver/remote/remote_connection_old.py	2024-07-18 21:04:29.951778163 +0200
+++ py/selenium/webdriver/remote/remote_connection.py	2024-07-18 21:05:28.084889592 +0200
@@ -301,7 +301,7 @@
         LOGGER.debug("%s %s %s", command_info[0], url, str(trimmed))
         return self._request(command_info[0], url, body=data)
 
-    def _request(self, method, url, body=None):
+    def _request(self, method, url, body=None, timeout=120):
         """Send an HTTP request to the remote server.
 
         :Args:
@@ -319,12 +319,12 @@
             body = None
 
         if self.keep_alive:
-            response = self._conn.request(method, url, body=body, headers=headers)
+            response = self._conn.request(method, url, body=body, headers=headers, timeout=timeout)
             statuscode = response.status
         else:
             conn = self._get_connection_manager()
             with conn as http:
-                response = http.request(method, url, body=body, headers=headers)
+                response = http.request(method, url, body=body, headers=headers, timeout=timeout)
             statuscode = response.status
         data = response.data.decode("UTF-8")
         LOGGER.debug("Remote response: status=%s | data=%s | headers=%s", response.status, data, response.headers)
